### ISSUE TITLE
Won't work with Microsoft SRP anymore. [x64 WIN 10]

### ISSUE THREAD

**Issue by hemiipatu** [dafzor/bnetlauncher#31]
Essentially, this used to work when Microsoft SRP was enabled and configured accordingly but for some reason - it's no longer the case. Interestingly, as soon as SRP is disabled the client works as per-normal.

I've attached the logs if that helps with future development. Also, would it be possible to provide any details on what and where the client is creating new files/folders etc. since this is usually what SRP is restricting.

`11:14:26.1569|10176|Program.Main:125|INFO|Starting bnetlauncher v2.1.6879.19645
11:14:26.1639|10176|Program.LogSystemInfo:706|INFO|Getting System details:
11:14:26.1699|10176|Program.LogSystemInfo:710|INFO|ENV: Microsoft Windows NT 10.0.17134.0 (4.0.30319.42000), 64bit
11:14:26.2269|10176|Program.LogSystemInfo:776|INFO|OS: Microsoft Windows 10 Enterprise (10.0.17134, 64-bit, 2057)
11:14:26.2319|10176|Program.LogSystemInfo:777|INFO|CPU: Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz, RAM: 16384MB
11:14:26.2369|10176|Program.LogSystemInfo:778|INFO|GPU: NVIDIA GeForce GTX 970 (25.21.14.1664, 4095MB)
11:14:26.2419|10176|Program.LogSystemInfo:779|INFO|HDD: SAMSUNG MZHPV128HDGM-00000, ST2000LM015-2E8174, Intel Raid 0 Volume
11:14:26.2469|10176|Program.LoadGameList:569|INFO|Loading gamedb files.
11:14:26.2549|10176|Program.LoadGameList:590|INFO|Loaded internal gamedb with '9' games.
11:14:26.2589|10176|Program.LoadGameList:607|INFO|Known games: '9'.
11:14:26.2629|10176|Program.Main:205|WARN|Ignoring unknown parameter: 'dst2'
11:14:26.2679|10176|Program.Main:213|INFO|Given parameter 'dst2'.
11:14:26.2718|10176|Program.Main:249|INFO|Using 'battlenet' client.
11:14:26.2768|10176|BnetClient.InstallPath:84|INFO|Client InstallPath:'C:\Program Files (x86)\Battle.net'.
11:14:26.2818|10176|Program.Main:265|INFO|Checking for other bnetlauncher processes using same client
11:14:26.2858|10176|Program.Main:273|INFO|No other bnetlauncher detected
11:14:26.2898|10176|BnetClient.Start:123|INFO|Starting 'battlenet' client.
11:14:26.2948|10176|BnetClient.InstallPath:84|INFO|Client InstallPath:'C:\Program Files (x86)\Battle.net'.
11:14:26.3198|10176|Tasks.Exists:125|INFO|Found task for battlenet.
11:14:26.3238|10176|Tasks.Run:92|INFO|Starting battlenet task.
11:14:28.3453|10176|BnetClient.WaitUntilReady:199|INFO|Waiting for battle.net client to be ready.
11:14:28.3973|10176|BnetClient.WaitUntilReady:206|WARN|Tried to WaitUntilReady with no battle.net client running.
11:14:28.4063|10176|Program.Main:309|INFO|Client 'Battle.net' not running and/or failed to start it.
11:14:39.3228|10176|Program.CloseClientIfLast:645|INFO|Closing client 'battlenet'.
11:14:39.3438|10176|Processes.KillProcessAndChildsById:40|WARN|Attempted`

**Comment by dafzor** [comment]
Did it stop working in version 2.00?

**Comment by hemiipatu** [comment]
I've just tested it now and it seems the release on August 4th (v.1.22) was the last version it would work on - So I believe that this issue was introduced in v2.00

**Comment by hemiipatu** [comment]
I also just had a look at what _might_ be blocking it (or the reason why) using sysinternals, but nothing obvious stood out at me. ie. It was being being blocked access to creating/accessing a file or folder etc.

Quite strange whatever is causing it.

**Comment by dafzor** [comment]
v2.00 introduced quite a few things:
- creation and use of schedule tasks to start the battle.net client (check task bnetlauncher\battlenet is created and active)
- use of a gamedb.ini database with configurations
- entirely new logging system.

Your log says it managed to create the scheduled task but then never actually starts it, could you see if you can launch the battle.net client by running it on demand?
![running-battlenet-task](https://user-images.githubusercontent.com/1612206/48640644-2bf86000-e9cf-11e8-91de-ac3b75451ba5.png)

Also can you provide more detail on what AppLocker policies are enabled on the system in question?

**Comment by hemiipatu** [comment]
So I have tried running Battle.net as per you would normally and everything runs perfectly fine - I then tried launching a game using Steam _(as described in the readme)_ and it failed, I tried to launch it directly from the task too, and that also failed.

The task is definitely there in Task scheduler and the status is _ready_.

**Comment by dafzor** [comment]
Could you show me the configuration of the task? I suspect  Microsoft SRP made bnetlauncher create the task as another user so when it tries to launch it it just fails due to it not being logged in.

If my guess is right you can fix it by manually changing the user in security options and changing the description to 9.99 to make sure newer versions of bnetlauncher do not try to update it.

**Comment by hemiipatu** [comment]
Apologies for the late reply, I've been super busy recently and _almost_ forgot about this.

I actually managed to get this resolved by making a few changes to the task manually. The changes involved included:
1. Run whether user is logged on or not
2. Configure for: Windows 10
3. If the task is already running > Stop the existing instance.

I also changed the formatting of the SRP rules that could affect the program and its' operation to:
1. %HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\ProgramFilesDir (x86)%
2. %HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\ProgramFilesDir (x86)%
3. %ProgramData%\Battle.net
4. E:\games\battle.net_library

Finally, since I also have 'Randomware protection' **on** I added both the bnetlauncher and Battle.net client on the whitelist.
